plugins/database/src/db_plugin.cpp

```diff
@@ -15454,13 +15448,46 @@ auto db_update_replica_access_time(irods::plugin_context& _ctx, const char* _inp
 
         std::size_t pos = 0;
         for (auto&& update : updates) {
-            stmt.bind(pos++, &update.at("atime").get_ref<const std::size_t&>());//.c_str());
-            stmt.bind(pos++, &update.at("data_id").get_ref<const std::size_t&>());//.c_str());
-            stmt.bind(pos++, &update.at("replica_number").get_ref<const std::size_t&>());//.c_str());
+            stmt.bind(pos++, update.at("atime").get_ref<const std::string&>().c_str());
+            stmt.bind(pos++, &update.at("data_id").get_ref<const std::size_t&>());
+            stmt.bind(pos++, &update.at("replica_number").get_ref<const std::size_t&>());
         }
 
         // Execute the batch of SQL statements.
         nanodbc::execute(stmt);
+#else
+        std::vector<std::size_t> data_ids;
+        std::vector<std::size_t> replica_numbers;
+        std::vector<std::string> access_times;
+
+        data_ids.reserve(updates.size());
+        replica_numbers.reserve(updates.size());
+        access_times.reserve(updates.size());
+
+        std::for_each(std::begin(updates), std::end(updates), [&data_ids, &replica_numbers, &access_times](const nlohmann::json& _j) {
+            data_ids.emplace_back(_j.at("data_id").get<std::size_t>());
+            replica_numbers.emplace_back(_j.at("replica_number").get<std::size_t>());
+            access_times.emplace_back(_j.at("atime").get<std::string>());
+        });
+
+        auto [db_instance, db_conn] = irods::experimental::catalog::new_database_connection();
+
+        constexpr const char* sql = "update R_DATA_MAIN set access_ts = ? where data_id = ? and data_repl_num = ?";
+        log_sql::debug("{}: SQL = [{}]", __func__, sql);
+
+        nanodbc::statement stmt{db_conn};
+        nanodbc::prepare(stmt, sql);
+
+        stmt.bind_strings(0, access_times);
+        stmt.bind(1, data_ids.data(), data_ids.size());
+        stmt.bind(2, replica_numbers.data(), replica_numbers.size());
+
+        // Execute the batch operation within a transaction and throw away the results.
+        // TODO Consider whether we should use execute() instead since the current code will
+        // rollback the updates on failure. The downside of using execute() is that it will
+        // have worst performance due to each update triggering an auto-commit instruction.
```

Comment:
```suggestion
        // have worse performance due to each update triggering an auto-commit instruction.
```

yes, we should count/log how often any failures happen... this will let us know whether we need to reconsider how we're updating.

Reply:
Done.